GoogleDataTransport/GDTCORLibrary/GDTCORMetricsController.m

```diff
@@ -24,32 +24,58 @@
 #import "FBLPromises.h"
 #endif
 
+#import "GoogleDataTransport/GDTCORLibrary/Internal/GDTCORRegistrar.h"
+
 @implementation GDTCORMetricsController
 
-- (void)storage:(nonnull id<GDTCORStoragePromiseProtocol>)storage
-    didDropEvent:(nonnull GDTCOREvent *)event {
-  // TODO(nickcooke): Implement.
++ (void)load {
+  [[GDTCORRegistrar sharedInstance] registerMetricsController:[self sharedInstance]
```

Comment:
Do we know why register two times with different target here?

Reply:
Yes. GDT's main components are singletons and registered on a per-target basis. Consider another main component like `GDTCCTUploader.m`, the shared instance is registered for several targets: https://github.com/google/GoogleDataTransport/blob/main/GoogleDataTransport/GDTCCTLibrary/GDTCCTUploader.m#L43-L49

Thanks for the question, I reached out to Firelog team to verify that these are the two targets that we want to register for.

Reply:
Update (after confirming with Firelog team): there are three target backends the SDK can send events to:
- Firelog
- Crashlytics
- Clearcut
(The SDK also has a unit test target and integration target for testing.)

For client metrics, we only want to collect metrics from events associated with Firelog and Crashlytics targets and upload those.